### ISSUE TITLE
Appear to have broken my firefox profile's tridactyl config

### ISSUE THREAD

**Issue by tuh8888** [tridactyl/tridactyl#2706]
*   Brief description of the problem:
Upon syncing to Firefor, I am unable to write commands with the `:` ex mode. Some keybindings do seem to be working like `f`, but by custom config `"~/.tridactylrc"` isn't loading. This makes Tridactyl unusable, which is a shame because it's so great when it's usable!

*   Steps to reproduce:

1. Starting with a fresh profile ("about:profiles" -> "create a new profile")
2. I open Firefox
3. Download the Tridactyl plugin from Firefox Addons
4. Tridactyl appears to install fine. My custom config loads correctly via `source ~/.tridactylrc` and all keybindings work, including using them on new tabs.
5. I sign in to my Firefox account to initiate Firefox sync
6. Tridactyl appears to reload (my custom theme is lost, leaving me with the default white mode indicator)
7. Opening a new tab, the mode indicator can't be seen and only a few keybindings work (`f` shows link hints). Notably, `:` does not bring up the ex mode so I can't issue any more commands to tridactyl.

*   Tridactyl version (`:version`):
1.19.1pre3928-53d0d18

*   Firefox version (Top right menu > Help > About Firefox): 
79.0 (64-bit)

*   URL of the website the bug happens on: 
All

*   Config (in a new tab, run `:viewconfig`, copy the url and paste it somewhere like gist.github.com): 

*   Contents of ~/.tridactylrc or ~/.config/tridactyl/tridactylrc (if they exist):

```
" General Settings
" set newtab about:blank
colorscheme breath

" Binds
bind r tabprev
bind s scrollline 10
bind f scrollline -10
bind t tabnext
bind a back
bind d forward

bind R scrollpx -50
bind S scrollline 1
bind F scrollline -1
bind T scrollpx 50
bind A followpage prev
bind D followpage next

bind ^ scrollto 0 x
bind $ scrollto 100 x

```

**Comment by hejops** [comment]
I've been noticing this kind of behaviour* after updating to FF79, but I just haven't been able to reproduce it consistently. Usually this happens whenever I make some minor changes to my config file and source it (which has gotten pretty unwieldy at this point). Creating a new profile (with only tridactyl enabled) usually produces the correct behaviour, but even then it might start to fail again after a few restarts.

It does work again when I restart my main profile a bunch of times. No idea, really.

*From memory, hjkl and hints work, but not `yank`/`source` or anything involving the command line. I think autocmds don't work too. FWIW, I disabled `fixamo` since I read it was deprecated.

**Comment by bovine3dom** [comment]
I think this is a duplicate of #2699 - in FF79+, the sync storage has a fairly tiny storage quota, smaller than most custom themes.

Plan A: if you don't have anything precious in your sync storage I'd run - in the address bar (!) - `tri sanitise tridactylsync`, remove the theme line from your RC and then restart.

Plan B would be to temporarily switch to Firefox ESR which doesn't have the quota yet.

Plan C would be to wait for me to merge #2665 and do a release : )

Sorry about this horrible bug. I knew it was coming but I didn't get round to fixing it in time :(

**Comment by dllud** [comment]
Bummer, it seems I lost all my Tridactyl settings due to this.

I updated to FF79 (FF78 was running). Upon restart Tridactyl lost all my settings. Looking at the modification time for `~/.mozilla/firefox/myprofile/extensions/tridactyl.vim@cmcaine.co.uk.xpi` hints that is was updated to `1.20.0` at around the same time I restarted FF. Perhaps I was running version Tridactyl version `1.19.1` when FF79 first started, which may have caused this issue?

Anyone got any suggestion on how I may recover my settings? Are they still stored somewhere on my Firefox local profile or the synced profile?
I do have a backup, but it is a couple of months old, and I had been doing some changes recently.

Anyway, lesson learnt: will use `~/.tridactylrc` from now on. No more trusting on FF.
